pkg/scm/git/git.go

```diff
@@ -217,83 +219,95 @@ func makePathAbsolute(source string) string {
 // expect git clone spec syntax; it also provides details if the file://
 // proto was explicitly specified, if we should use OS copy vs. the git
 // binary, and if a frag/ref has a bad format
-func ParseFile(source string) (details *FileProtoDetails, mods *URLMods) {
+func ParseFile(source string) (details *FileProtoDetails, mods *URLMods, err error) {
+	// NOTE: Why don't we just trim the source here?
```

Comment:
You understood correctly, I was wondering what the use-cases for file:// were.

Within ParseFile, whenever we use `path` we trim the prefix before using it.  If we trimmed it at the beginning, we wouldn't have to make those checks further down.  

Since the function name states we're working with a file, would it make sense to just trim the file:// prefix outright?  We'd still be tracking its use with `protoSpecified`